libctl/structure.cpp

```diff
@@ -1300,8 +1297,18 @@ static meep::susceptibility *make_multilevel_sus(const multilevel_atom *d) {
   for (int t = 0; t < d->transitions.num_items; ++t) {
     int i = d->transitions.items[t].from_level - minlev;
     int j = d->transitions.items[t].to_level - minlev;
-    Gamma[i*L+i] += d->transitions.items[t].transition_rate;
-    Gamma[j*L+i] -= d->transitions.items[t].transition_rate;
+    Gamma[i*L+j+1] -= d->transitions.items[t].transition_rate;
+    Gamma[(i-1)*L+j+1] += d->transitions.items[t].transition_rate;
```

Comment:
Gamma[i*L+j+1] and Gamma[(i-1)*L+j+1] are entries with opposite signs in rows one below the other which thus ensures that the rows sum to zero. The previous version had Gamma[i*L+i] and Gamma[j*L+i] where the rows didn't sum to zero. The entry and sign convention for Gamma is based on eqns. 2-5 of Song's 2011 Optics Express paper.